src/ConnectedMode/Binding/ImportBeforeFileGenerator.cs

```diff
@@ -0,0 +1,105 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using System.Text;
+using System.Xml;
+
+namespace SonarLint.VisualStudio.ConnectedMode.Binding
+{
+    internal class ImportBeforeFileGenerator
+    {
+        private readonly IFileSystem fileSystem;
+
+        private const string targetsFileName = "SonarLint.targets";
+
+        public ImportBeforeFileGenerator() : this(new FileSystem()) { }
+
+        public ImportBeforeFileGenerator(IFileSystem fileSystem)
+        {
+            this.fileSystem = fileSystem;
+
+            CreateTargetsFileIfNotExists();
+        }
+
+        private void CreateTargetsFileIfNotExists()
```

Comment:
The logic in the two methods `CreateTargetsFileIfNotExists` and `WriteFile` is fine, but the way it is split between the two methods seems a bit odd.

In spite of it's name, the `CreateTargetsFileIfNotExists` doesn't actually do any of the checking for directory/file existence - it's all done in `WriteFile`.